src/useForm.ts

```diff
@@ -624,9 +623,10 @@ function useForm(form?: FormInstance): [FormInstance] {
         forceUpdate({});
       };
 
-      const formStore: FormStore = new FormStore(forceReRender);
+      const formStore: FormStore<T> = new FormStore<T>(forceReRender);
 
-      formRef.current = formStore.getForm();
+      // @NOTE getForm() and formRef types do not match, bug?
```

Comment:
One is `InternalFormInstance` and another is `FormInstance`.

Reply:
Indeed, but the typings don't match, as `FormInstance` seems to be a subset of `InternalFormInstance`, so either `.getForm()` should return a `FormInstance` or typing of the ref should be `InternalFormInstance` or there would be a need of some casting function like:

```ts
const asPublicFormInstance = (form: InternalFormInstance): FormInstance => pick(form, ...);
```

Reply:
> One is InternalFormInstance and another is FormInstance.

It's describe why `@NOTE getForm() and formRef types do not match, bug?`. We only export the FormInstance with TS since InternalFormInstance is not promise that we will provide.